mlir/lib/CAPI/CMakeLists.txt

```diff
@@ -1,7 +1,30 @@
 add_mlir_public_c_api_library(QuantumCAPI
+    CompilerDriver.cpp
     Dialects.cpp
 
     LINK_LIBS PRIVATE
+    CatalystCompilerDriver
+    MLIRCatalyst
+    MLIRCatalystTransforms
     MLIRQuantum
+    quantum-transforms
     MLIRGradient
+    gradient-transforms
+
+    MhloRegisterDialects
+    StablehloRegister
+
+    # Taken from mlir-hlo/bindings/c/CMakeLists.txt:
+    # For AllMhLoPasses:
+    ChloPasses
+    MhloPasses
+    MhloToLhloConversion
+    MhloToArithmeticConversion
+    MhloToMemrefConversion
+    MhloToStandard
+    MhloToThloConversion
+    MhloToLinalg
+    MhloToStablehlo
+    MhloShapeOpsToStandard
+    StablehloToMhlo
```

Comment:
From the maintenance pov, wasn't there any way to not hardcoding all MHLO passes?

Reply:
There is an `AllMhloPasses` interface library, but this unfortunately doesn't seem to be exported in the MHLO build. I'll move these to a top-level variable.